src/components/partnerships/PartnershipsMap.tsx

```diff
@@ -13,82 +13,87 @@ type PartnerShipsMapProps = {
-          countryFocus: partnershipconfig?.iso3,
-        }}
-        studyPinsConfig={{
-          records: records,
-        }}
-      />
-  );
-};
-
 export default function PartnerShipsMap({ partnershipconfig }: PartnerShipsMapProps) {
   const [state] = useContext(AppContext);
   const [records, setRecords] = useState<AirtableRecord[]>([]);
   const [estimateGradePrevalence, setEstimateGradePrevalence] = useState<EstimateGradePrevalence | undefined>();
+  const [isLoading, setIsLoading] = useState<boolean>(false);
 
   useEffect(() => {
     const api = new httpClient();
     const getPartnershipsData = async () => {
+      setIsLoading(true);
       const { records, estimateGradePrevalences } = await api.getCountryPartnershipData({country: [partnershipconfig?.routeName]});
       setRecords(records);
       setEstimateGradePrevalence(estimateGradePrevalences[0]);
+      setIsLoading(false);
     }
-    getPartnershipsData()
+    getPartnershipsData();
   }, [partnershipconfig, state.explore.estimateGradePrevalences])
   const isMobileDeviceOrTablet = useMediaQuery({ maxDeviceWidth: mobileDeviceOrTabletWidth })
 
   if (partnershipconfig === undefined) {
     return <></>;
   }
 
+  const desktopMap = (
+    partnershipconfig: Partnership,
+    estimateGradePrevalence: EstimateGradePrevalence | undefined,
+    records: AirtableRecord[]
+  ) => {
+    return (
+      <>
+        <div className="col-9 h-100">
+          <Loader indeterminate active={isLoading}/>
+          <MapboxMap
+            mapConfig={partnershipconfig.mapboxMapOptions}
+            countriesConfig={{
+              estimateGradePrevalences: estimateGradePrevalence ? [estimateGradePrevalence] : undefined,
+              countryFocus: partnershipconfig?.iso3,
+            }}
+            studyPinsConfig={{
+              records: records,
+            }}
+          />
+        </div>
+        <div className="col-3 h-100">
+          <Legend hideLayers />
+            <div className="mt-1">{Translate("PartnershipsView", ["MapNote1"])}</div>
+            <div className="mt-1">
+                {Translate("PartnershipsView", ["MapNote2"], {
+                    WEBSITE: "ourworldindata.org",
+                })}
+            </div>
+        </div>
+      </>
+    );
+  };
+  
+  // Note: Loading spinner doesn't work very well on mobile
+  // Not a priority to implement because mobile is deprecated anyways
+  const mobileMap = (
+    partnershipconfig: Partnership,
+    estimateGradePrevalence: EstimateGradePrevalence | undefined,
+    records: AirtableRecord[]
+  ) => {
+    return (
+        <MapboxMap
+          mapConfig={partnershipconfig.mapboxMapOptions}
+          countriesConfig={{
+            estimateGradePrevalences: estimateGradePrevalence ? [estimateGradePrevalence] : undefined,
+            countryFocus: partnershipconfig?.iso3,
+          }}
+          studyPinsConfig={{
+            records: records,
+          }}
+        />
+    );
+  };
+
```

Comment:
What is the reasoning behind moving these functions from outside to inside the component?